internal/credential/repository_credentiallibrary.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// A CredentialLibraryRepository provides read only information
+// generic to all credential store types.
+type CredentialLibraryRepository struct {
+	reader db.Reader
+}
+
+// NewCredentialLibraryRepository creates a new CredentialLibraryRepository.
+func NewCredentialLibraryRepository(ctx context.Context, r db.Reader) (*CredentialLibraryRepository, error) {
+	const op = "credential.NewCredentialLibraryRepository"
+	switch {
+	case r == nil:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "db.Reader")
+	}
+
+	return &CredentialLibraryRepository{
+		reader: r,
+	}, nil
+}
+
+// Now returns the current timestamp in the DB.
+func (clr *CredentialLibraryRepository) Now(ctx context.Context) (time.Time, error) {
```

Comment:
Will we need to repeat this code in other repositories?

Reply:
Yes, we repeat this in a lot of different repositories. Do you think we should make a repository for just this part that we could use in the handlers?